python/understack-workflows/understack_workflows/nautobot_device.py

```diff
@@ -139,26 +138,26 @@ def switches_for(nautobot, chassis_info: ChassisInfo) -> dict:
 def nautobot_switches(nautobot, mac_addresses: set[str]) -> dict[str, dict]:
     """Get switches by MAC address.
 
+    Assumes switch MAC addresses are present in Nautobot in a custom field on
+    Device called chassis_mac_address.
+
     Assumes that MAC addresses in Nautobot are normalized to upcase
     AA:BB:CC:DD:EE:FF form.
 
-    We store the MAC address in the nautobot device "asset tag" field because
-    there was nowhere else to put it.
-
     returns a dict[mac_address] -> dict switch information indexed by mac
     """
-    formatted_list = json.dumps(list(mac_addresses))
+    pattern = "|".join(mac_addresses)
 
     query = (
         """{
-        devices(asset_tag: %s){
+        devices(cf_chassis_mac_address__re: "(%s)"){
```

Comment:
Do we have to do a regex? isn't that going to be a slow query?

Reply:
We had to change how we do the graphql filter - you could search in Nautobot's built-in string attributes by passing an array of possible values to find, but that doesn't work for custom fields.   Regexp does work, however.

I assumed that this is somehow translated to SQL and executed by the database but I didn't prove that.  The performance is good on our staging system, and certainly much faster than performing multiple individual queries.  Not sure how else this can be done.

Reply:
Well, I did have the idea of moving a lot of this code to the Nautobot plugin, but that doesn't quite resonate with me for some reason.

Reply:
ah okay. I was just asking about the regex. I'm a +1 on less plugin. I'm hoping we can get to a clean install at some point.